### ISSUE TITLE
BatchNormalization

### ISSUE THREAD

**Issue by shuzi** [soumith/cudnn.torch#200]
Hi,  does the   BatchNormalization has special treatment about zero?
For text application we often have zero paddings and those padded zeros should not be used when calculating the mean the standard deviation.

**Comment by ngimel** [comment]
No, it does not. You should remove your explicit padding before passing data to batchNormalization. Try to keep batchNorm input contiguous - performance is much better this way, so this might mean you'll have to copy data rather than play with the views and strides.
